exercises/Stephen_Smith/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,40 @@
+/**
+ * Skeleton main routine
+ */
+
+#include "../include/ShapeSorter.h"
+
+
+using namespace std;
+int main(int, char **)
+{
+
+    // Create shapes
+    auto sCircle_1 = make_shared<Circle>(1.5);
```

Comment:
It is fine to use `shared_ptr`, it's good that the memory is managed by something. A first point of call however should be `std::unique_ptr`. This is a simpler wrapper that makes it much clearer who owns a particular type as there is no sharing possible and is just a basic wrapper around `new`/`delete`.

Reply:
Okay I'll keep that in mind. For now I've switched to using raw pointers referencing the stack generated objects. I need to do a bit more reading around these smart pointers.